lib/block_view.go

```diff
@@ -5016,3 +5033,10 @@ func mergeExtraData(oldMap map[string][]byte, newMap map[string][]byte) map[stri
 
 	return retMap
 }
+
+func (bav *UtxoView) GetMaxBlockSizeBytes(height uint64) uint64 {
```

Comment:
IMO, this has a good chance of becoming error prone in the future. Imagine if someone instantiates a UtxoView at block height n, which uses the global params at height n, but passes a height of n+2 to this function.

Ideally, the PoS code just reads `bav.GetCurrentGlobalParamsEntry().MaxBlockSizeBytesPoS` directly where needed. Direct access is fine in this case

Reply:
Thinking more about it - I think we want to use the value from a snapshot anyways. we really just need to distinguish between PoS block size and PoW block size.